File: javascript/discordjs-v14/scripts/slashcommands/moderação/banir.js
//Command ban
//Exemplo: /ban <usuário> <motivo>
//Linguagem usada: js
//Author: aquelemesmoojack#4306
//Versão: Discord.JS v14

const { ApplicationCommandType, ApplicationCommandOptionType, PermissionBitField } = require("discord.js")

module.exports = {
    name: "banir",
    description: "Bane um usuário",
    type: ApplicationCommandType.ChatInput,
    options: [
        {
            name: "membro",
            description: "Mencione o membro que você quer banir",
            type: ApplicationCommandOptionType.User,
            required: true
        },
        {
            name: "motivo",
            description: "Insira o motivo",
            type: ApplicationCommandOptionType.String,
            required: true
        }
    ],
    run: async (client, interaction) => {
        if(interaction.member.permission.has(PermissionBitField.Flags.BanMembers)) {
            const membro = interaction.options.getUser("membro")
            const motivo = interaction.options.getString("motivo")

            interaction.reply({content: "Membro banido com sucesso", ephemeral: true})
            membro.ban({reason: [motivo]})
        } else {
            return interaction.reply({content: "Você não tem permissão", ephemeral: true})
        }
    }
}

File: python/scripts/moderação/banir.py
# Command ban
# Exemplo: /ban <usuário> <motivo>
# Linguagem usada: py
# Author: Luis Gomes

import discord
from discord.ext import commands
from discord import app_commands

intents = discord.Intents.all()
bot = commands.Bot(command_prefix='!', intents=intents)

@bot.event
async def on_ready():
    print(f'Bot conectado como {bot.user}')
    try:
        synced = await bot.tree.sync()
        print(f'Slash commands sincronizados: {len(synced)} comandos')
    except Exception as e:
        print(f'Erro ao sincronizar slash commands: {e}')

@bot.event
async def on_command_error(ctx, error):
    if isinstance(error, commands.CommandNotFound):
        await ctx.send('Comando não encontrado.')
    else:
        await ctx.send(f'Ocorreu um erro: {error}')

@bot.tree.command(name='banir', description='Banir um usuário')
@commands.has_permissions(ban_members=True)
@app_commands.describe(usuario='Usuário que será banido', motivo='Motivo do banimento')
async def ban(interaction: discord.Interaction, usuario: discord.Member, motivo: str):
    if interaction.user.top_role <= usuario.top_role:
        await interaction.response.send_message('Não podes banir alguém com um cargo maior ou igual ao teu.')
        return

    try:
        await usuario.ban(reason=motivo)
        await interaction.response.send_message(f'{usuario.mention} foi banido. Motivo: {motivo}')
    except discord.Forbidden:
        await interaction.response.send_message('Não tenho permissão para banir esse usuário.')
    except discord.HTTPException:
        await interaction.response.send_message('Ocorreu um erro ao tentar banir o usuário.')
    except Exception as error:
        await interaction.response.send_message(f'Erro ao tentar banir: {error}')

bot.run('TOKEN')
